src/dataflow/src/source/mod.rs

```diff
@@ -966,6 +966,10 @@ where
             let mut bytes_read = 0;
             // Accumulate updates to offsets for system table metrics collection
             let mut metric_updates = HashMap::new();
+            // System table metrics will be written either when we have caught up with the
+            // source or when we have seen METRIC_FLUSH_INTERVAL number of messages
+            const METRIC_FLUSH_INTERVAL: i32 = 1000;
```

Comment:
Is there any reasoning behind this 1000 number?

Reply:
None whatsoever, other than it was biggish. I couldn't think of a great way of coming up with anything concrete short of seeding a bunch of data into kafka and benchmarking bootstrapping - which seemed excessive. Open to suggestions though 🙂

Reply:
Meh, it sounds fine to me!